sdk/README.md

```diff
@@ -0,0 +1,334 @@
+	component.RunnerWithComponentName("sample-target"),
+)
+```
+
+### Components
+
+#### Target
+
+A `Target` component should be used to prepare a target for scanning.
+
+For example, cloning a repository and make it available for a `Scanner` to scan.
+
+You can create a new `Target` component like follows:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/smithy-security/smithy/sdk/component"
+)
+
+type sampleTarget struct{}
+
+func (s sampleTarget) Close(ctx context.Context) error {
+	// Close your component here!
+	return nil
+}
+
+func (s sampleTarget) Prepare(ctx context.Context) error {
+	// Prepare the target here!
+	return nil
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	if err := component.RunTarget(ctx, sampleTarget{}); err != nil {
+		log.Fatalf("unexpected run error: %v", err)
+	}
+}
+```
+
+#### Scanner
+
+A `Scanner` scans a `Target` to find vulnerabilities.
+
+You can create a new `Scanner` component like follows:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+type (
+	sampleScanner struct{}
+
+	sampleRawVuln struct{}
+)
+
+func (s sampleRawVuln) Unmarshal() (*ocsf.VulnerabilityFinding, error) {
+	// Tell us how to convert your payload to ocsf format here!
+	return &ocsf.VulnerabilityFinding{}, nil
+}
+
+func (s sampleScanner) Close(ctx context.Context) error {
+	// Close your component here!
+	return nil
+}
+
+func (s sampleScanner) Store(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error {
+	// Store your findings here!
+	return nil
+}
+
+func (s sampleScanner) Scan(ctx context.Context) ([]component.Unmarshaler, error) {
+	// Scan a target and return a payload here!
+	return nil, nil
+}
+
+func (s sampleScanner) Transform(ctx context.Context, payload component.Unmarshaler) (*ocsf.VulnerabilityFinding, error) {
+	// Transform your payload to ocsf format!
+	return &ocsf.VulnerabilityFinding{}, nil
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	if err := component.RunScanner(ctx, sampleScanner{}); err != nil {
+		log.Fatalf("unexpected run error: %v", err)
+	}
+}
+```
+
+#### Enricher
+
+An `Enricher` annotates vulnerability findings with extra information.
+
+You can create a new `Enricher` component like follows:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+type sampleEnricher struct{}
+
+func (s sampleEnricher) Close(ctx context.Context) error {
+	// Close your component here!
+	return nil
+}
+
+func (s sampleEnricher) Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error) {
+	// Read the prepared vulnerability reports here.
```

Comment:
Does this Read from the database or any intermediate data store?
If so, in reality does it call the sdk method "Read"?
If so, does it make sense to be called Load?

Reply:
I don't feel strongly about the name

This will fetch ready VulnerabilityFindings from some source. By default this will be from SQLite and on SAAS from another data source - but one can override it - for example, reading from s3 and such